server/src/org/apache/cloudstack/ca/CAManagerImpl.java

```diff
@@ -328,23 +330,30 @@ protected void runInContext() {
                     final String hostDescription = String.format("host id=%d, uuid=%s, name=%s, ip=%s, zone id=%d",
                             host.getId(), host.getUuid(), host.getName(), hostIp, host.getDataCenterId());
 
-                    try {
-                        certificate.checkValidity(now.plusDays(CertExpiryAlertPeriod.valueIn(host.getClusterId())).toDate());
-                    } catch (final CertificateExpiredException | CertificateNotYetValidException e) {
+                    Date notAfterDate = certificate.getNotAfter();
+                    DateTime notAfter = new DateTime(notAfterDate);
+                    DateTime alertDate = notAfter.minusDays(CertExpiryAlertPeriod.valueIn(host.getClusterId()));
+                    DateTime warnDate = alertDate.minusDays(CertExpiryWarningPeriod.valueIn(host.getClusterId()));
+
+                    if (now.isAfter(alertDate)) {
                         LOG.warn("Certificate is going to expire for " + hostDescription);
                         if (AutomaticCertRenewal.valueIn(host.getClusterId())) {
                             try {
                                 LOG.debug("Attempting certificate auto-renewal for " + hostDescription);
                                 boolean result = caManager.provisionCertificate(host, false, null);
                                 if (result) {
                                     LOG.debug("Succeeded in auto-renewing certificate for " + hostDescription);
+                                    caManager.sendAlert(host, "Certificate auto-renewal succeeded for " + hostDescription,
+                                            "Certificate auto-renew succeeded for " + hostDescription);
                                 } else {
                                     LOG.debug("Failed in auto-renewing certificate for " + hostDescription);
```

Comment:
we should make this LOG.warn. This would match line 354.  (I know it was debug in the original code)

Reply:
That makes sense to me. I've made the change and pushed it up.